### PR TITLE
Generate User, Category, UserCategory models

### PR DESCRIPTION
**Pull request by PandaGrom**
# Description
[Task](https://trello.com/c/O4KOOz1t/41-generate-user-category-usercategory-models)

- Create User model (Devise)
- Create Category model
- Create UserCategory model
- Add aasm status
- Add avatar for User model (Active Storage)

## Type of change

- [x] New feature

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2022_01_18_152931) do
```
Comment (nieisnuje) [inline db/schema.rb]: schema was created in another PR
merge changes from this PR to your branch
and add your changes to the existing schema

File: spec/models/category_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (nieisnuje) [inline spec/models/category_spec.rb]: write tests

Comment (PandaGrom) [reply spec/models/category_spec.rb]: Fixed!

File: spec/support/factory_bot.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
```
Comment (nieisnuje) [inline spec/support/factory_bot.rb]: add empty line

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+class PhoneNumberValidator < ActiveModel::Validator
+  def validate(record)
+    return if Phonelib.valid? record.phone_number
+
+    record.errors.add :phone_number, 'Invalid telephone number'
+  end
+end
```
Comment (nieisnuje) [inline app/models/user.rb]: create `validators` folder and move it there

**Review by nieisnuje — COMMENTED** [review]
`tmp` folder is in `gitignore` now, you should not have files from it in your pr
merge main into your branch

**Review by nieisnuje — APPROVED** [review]
you still haven't done anything about schema.rb
your teammates will have to resolve git conflicts because of it

**Review by nieisnuje — APPROVED** [review]

**Review by AndrewTehan — APPROVED** [review]

**Review by alhimick-prog — APPROVED** [review]

**Pull request merged**